Shared library can't be deleted on shutdown in Windows

When the shutdown hook from [`Util:51`](https://github.com/electronstudio/jaylib-ffm/blob/c8218ae951fe5aa8cbd6b69ace2b7747b928648b/src/main/java/uk/co/electronstudio/Util.java#L51) runs on my Windows 11 system, it results in an `AccessDeniedException` as shown below, and the raylib dll that was extracted to the user's temp directory isn't deleted. When iterating on project code this results in a lot of duplicate dlls accumulating in the temp directory.

Example stacktrace:
```
Exception in thread "Thread-0" java.lang.RuntimeException: java.nio.file.AccessDeniedException: C:\Users\USERNAME\AppData\Local\Temp\6737253390355114103.dll
        at uk.co.electronstudio.Util$1.run(Util.java:53)
Caused by: java.nio.file.AccessDeniedException: C:\Users\USERNAME\AppData\Local\Temp\6737253390355114103.dll
        at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:89)
        at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
        at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108)
        at java.base/sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:273)
        at java.base/sun.nio.fs.AbstractFileSystemProvider.delete(AbstractFileSystemProvider.java:104)
        at java.base/java.nio.file.Files.delete(Files.java:1152)
        at uk.co.electronstudio.Util$1.run(Util.java:51)
```

I spent a bit of time digging into this to see if I could sort out the cause since it seems odd that `Files.createTempFile(...)` would allow a process to write a file to the temp directory but not allow the same process to delete it. While I didn't come up with a definite explanation, my guess is that the shared library isn't unloaded at the time the shutdown hook runs.

The FFM API is new for me, but reviewing the library code I wonder if a solution might involve changing `raylib_h_1.LIBRARY_ARENA` from `Arena.ofAuto()` to a confined arena that can be managed directly rather than automatically, such that it could be manually closed either as a part of the shutdown hook before trying to delete the shared lib file, or via an explicit `Raylib.shutdown()` call made by application code.

--- a/src/main/java/uk/co/electronstudio/Util.java
+++ b/src/main/java/uk/co/electronstudio/Util.java
@@ -4,12 +4,14 @@ import java.io.IOException;
 import java.io.InputStream;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.nio.file.StandardCopyOption;
 
 public class Util {
 
     public static final String OS_ARCH = System.getProperty("os.arch");
     public static final String OS_NAME = System.getProperty("os.name");
+    public static final String TMP_DIR = System.getProperty("java.io.tmpdir");
     public static final boolean IS_OS_LINUX = OS_NAME.startsWith("Linux") || OS_NAME.startsWith("LINUX");
     public static final boolean IS_OS_LINUX_AMD64 = IS_OS_LINUX && OS_ARCH.startsWith("amd64");
     public static final boolean IS_OS_MAC = OS_NAME.startsWith("Mac");
@@ -28,9 +30,12 @@ public class Util {
     }
 
     public static String extractFileFromResources(String name, String extension) {
-
         try {
-            Path extractedLoc = Files.createTempFile(null, extension).toAbsolutePath();
+            // ensure that the library's temp directory exists in the system temp dir
+            Path libraryTempDir = Paths.get(TMP_DIR, "jaylib-ffm");
+            Files.createDirectories(libraryTempDir);
+
+            // get specified file resource from library jar
             String path = "/" + name + extension;
             InputStream source = null;
             try {
@@ -43,18 +48,11 @@ public class Util {
                 throw new RuntimeException("Couldn't extract "+name+extension+" from resources");
             }
 
+            // copy file into libraryTempDir, overwriting if it already exists
+            Path extractedLoc = libraryTempDir.resolve(name+extension);
             Files.copy(source, extractedLoc, StandardCopyOption.REPLACE_EXISTING);
 
-            Runtime.getRuntime().addShutdownHook(new Thread() {
-                public void run() {
-                    try {
-                        Files.delete(extractedLoc);
-                    } catch (IOException e) {
-                        throw new RuntimeException(e);
-                    }
-                }
-            });
-
+            // absolute path to extracted file
             return extractedLoc.toString();
         } catch (IOException e) {
             throw new RuntimeException(e);

